src/main/java/com/example/petbuddybackend/utils/annotations/validation/DateRangeValidator.java

```diff
@@ -27,14 +30,22 @@ public boolean isValid(Object value, ConstraintValidatorContext context) {
             startDateField.setAccessible(true);
             endDateField.setAccessible(true);
 
-            Comparable<Object> startDate = (Comparable<Object>) startDateField.get(value);
-            Comparable<Object> endDate = (Comparable<Object>) endDateField.get(value);
+            Object startDate = startDateField.get(value);
+            Object endDate = endDateField.get(value);
 
             if(startDate == null || endDate == null) {
                 return true;
             }
 
-            boolean isValid = startDate.compareTo(endDate) < 0;
+            if(!(startDate instanceof Temporal) || !(endDate instanceof Temporal) ||
+                    !(startDate instanceof Comparable) || !(endDate instanceof Comparable)) {
+                throw new IllegalActionException("DateRangeValidator can only be used on Date fields implementing Comparable");
+            }
+
+            Comparable<Object> startDateComparable = (Comparable<Object>) startDate;
+            Comparable<Object> endDateComparable = (Comparable<Object>) endDate;
+
+            boolean isValid = startDateComparable.compareTo(endDateComparable) < 0;
```

Comment:
W sumie isValid to może być funkcja osobna

```java
    private static boolean isValid(Object startDate, Object endDate) {
        if(!(startDate instanceof Temporal) || !(endDate instanceof Temporal) ||
                !(startDate instanceof Comparable) || !(endDate instanceof Comparable)) {
            throw new IllegalActionException("DateRangeValidator can only be used on Date fields implementing Comparable");
        }

        Comparable<Object> startDateComparable = (Comparable<Object>) startDate;
        Comparable<Object> endDateComparable = (Comparable<Object>) endDate;

        return startDateComparable.compareTo(endDateComparable) < 0;
    }
```

Reply:
w sumie